pallets/gear/src/tests.rs

```diff
@@ -6261,6 +6264,83 @@ fn exit_locking_funds() {
     });
 }
 
+#[test]
+fn frozen_funds_remain_on_exit() {
+    use crate::{fungible, Fortitude, Preservation};
+    use demo_constructor::{Calls, Scheme};
+    use frame_support::traits::{LockableCurrency, WithdrawReasons};
+
+    init_logger();
+    new_test_ext().execute_with(|| {
+        let (_init_mid, program_id) = init_constructor(Scheme::empty());
+
+        let program_account = program_id.cast();
+
+        let user_2_initial_balance = CurrencyOf::<Test>::free_balance(USER_2);
+
+        assert!(Gear::is_initialized(program_id));
+
+        // Funding program account with this amount
+        let value = 5_000;
+
+        let calls = Calls::builder().send_value(program_id.into_bytes(), [], value);
+        assert_ok!(Gear::send_message(
+            RuntimeOrigin::signed(USER_1),
+            program_id,
+            calls.encode(),
+            10_000_000_000,
+            value,
+            false,
+        ));
+
+        run_to_next_block(None);
+
+        // free balance includes the ED
+        assert_eq!(CurrencyOf::<Test>::free_balance(program_account), value);
+
+        // reducible balance for preserved account doesn't include the ED
```

Comment:
program should be able to operate of the ED.
e.g. 
ed = 10
program has 15, while 2 is frozen
program must be able to send 13

Reply:
test it pls

Reply:
if the idea is to always keep a program's account alive then it won't be able to touch the ED (provided we get the `reducible_balance` with `Preservation::Preserve` parameter. Otherwise we can pass the `Preservation::Expandable` as the preservation requirement, thereby allowing the program's account to be wiped as dust in case the balance drops below the ED